inst/rsn/RSN_01.Rmd

```diff
@@ -0,0 +1,229 @@
+pts_ppp<-ppp(pts@coords[,1], pts@coords[,2], window=egypt)
+dlet<-dirichlet(pts_ppp)
+
+owin2Polygons <- function(x, id="1") {
+  stopifnot(is.owin(x))
+  x <- as.polygonal(x)
+  closering <- function(df) { df[c(seq(nrow(df)), 1), ] }
+  pieces <- lapply(x$bdry,
+                   function(p) {
+                     Polygon(coords=closering(cbind(p$x,p$y))) })
+  z <- Polygons(pieces, id)
+  return(z)
+}
+
+tess2SP <- function(x) {
+  stopifnot(is.tess(x))
+  y <- tiles(x)
+  nom <- names(y)
+  z <- list()
+  for(i in seq(y))
+    z[[i]] <- owin2Polygons(y[[i]], nom[i])
+  return(SpatialPolygons(z))
+}
+
+voronoi<-tess2SP(dlet)
+
+```
+
+### Randomly sample within Voronoi polygons
+
+```{r samp_in_voronoi, message=FALSE, warning=FALSE}
+
+allpts=NULL
+
+for(i in 1:length(voronoi)){
+  sampledpts<-spsample(voronoi[i],n=2,type="random")
+    if(i==1){
+      allpts=sampledpts
+    } else{
+      allpts<-rbind(allpts, sampledpts)
+    }
+}
+allpts<-SpatialPoints(data.frame(allpts))
+allpts$id<-1:nrow(allpts@coords)
+
+plot(egypt, main=NULL); 
+plot(voronoi, add=TRUE, main=NULL);
+plot(allpts, add=TRUE, main=NULL)
+```
+
+### Build Watts-Strogatz model
+
+First, we build a Watts-Strogatz small-world model based on the derived points. We will also assign it a random farm size value between 1 and 100.
+
+```{r}
+
+ws<-watts.strogatz.game(1, nrow(allpts@coords), 5, 0.05)
+V(ws)$farmsize<-round(runif(length(V(ws)), 1, 100))
+
+```
+
+### Build "real-world" model
+
+We can also use the generated data to create another model closer to the real world. We begin by assuming that there is a lesser likelihood that points less than 50 km apart would be connected--individual farmers will likely not travel >50 km to sell chickens. Then, to assign some randomness to points over large distances, we randomly create links between points, simulating to the Strogatz model.
+
+```{r}
+
+distMat<-round(spDists(as.matrix(allpts@coords), longlat=TRUE))  # dist in km
+rownames(distMat)<-allpts$id
+
+# Process KNOWN
+
+distMat[distMat > 50] <- 0  # No links between points > 50 km apart
+mat1<-setNames(melt(distMat), c('row', 'col', 'values'))
+mat1$values[mat1$values != 0] <- 1  # all other links are 1
+
+# Now let's randomly rewire some 0s to 1s
+
+mat1$newvalues<-sample(c(0,1), size=nrow(mat1), replace=TRUE, prob=c(.99, 0.01))
+
+# Now we remove all the matrices where we didn't have any links, and pick out only # the connected network
+
+mat2<-mat1[mat1$newvalues==1,]; mat2$values<-NULL;
+
+# We will also set up vertices here
+nodes<-append(unique(mat2$row), unique(mat2$col))
+nodes<-data.frame(row=unique(nodes))
+
+# Randomly assign a "household" anywhere between 1-100 chickens
+
+nodes$farmsize<-round(runif(nrow(nodes), 1, 100))
+
+rw <- graph_from_data_frame(d=mat2, vertices=nodes, directed=T)
+
+```
+
+This function derives network and community characteristics from the points. The outputs are added to the global environment with a prefix of choice
+
```

Comment:
An interesting strategy! Fine for here but I would generally return these as a list and avoid such side-effects as its often unexpected behavior.  I prefer to write functions anticipating they might be incorporated into a package at a later time.